lib/widgets/medication_details.dart

```diff
@@ -0,0 +1,670 @@
+                    items: (items)
+                        .map<DropdownMenuItem<String>>((vt) =>
+                            DropdownMenuItem<String>(
+                                value: vt['name'],
+                                child: Text(vt['name'] ?? 'unknown')))
+                        .toList(),
+                    validator: (value) => value == null ? dosingInstructionRequired : null,
+                  );
+                },
+                valueListenable: _typeChangeAllowed,
+              ))
+        ]));
+  }
+
+  Padding _numericRow(String lblTitle, TextEditingController controller) {
+    return Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 16.0),
+        child:
+            Row(mainAxisAlignment: MainAxisAlignment.spaceBetween, children: [
+          Text(lblTitle),
+          Row(
+            children: [
+              Container(
+                padding: EdgeInsets.only(right: 10),
+                alignment: Alignment.centerRight,
+                width: 50,
+                child: TextFormField(
+                    keyboardType: TextInputType.number,
+                    controller: controller,
+                    validator: (value) => value!.isEmpty ? controller == _doseController ? doseRequired : durationRequired : null,
+                    onTapOutside: (_) {
+                      setState(() {
+                        if (controller == _doseController) {
+                          _medication.dosingInstructions?.dose =
+                              double.tryParse(_doseController.text);
+                        } else {
+                          _medication.duration =
+                              int.tryParse(_durationController.text);
+                        }
+                        _updateResult();
+                      });
+                    }
+                    ),
+              ),
+              lblTitle=='Dose'? _unitRow(_dosageInstructions?['doseUnits']) : _durationUnitRow(_dosageInstructions?['durationUnits'])
+            ],
+          ),
+        ]));
+  }
+
+  Padding _toggleRow(String lblTitle) {
+    return Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 16.0),
+        child:
+        Row(mainAxisAlignment: MainAxisAlignment.spaceBetween, children: [
+          Text(lblTitle),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              Padding(
+                padding: EdgeInsets.symmetric(horizontal: 3.0),
+                child: SizedBox(
+                  width: 30,
+                  height: 50,
+                  child: TextFormField(
+                    keyboardType: TextInputType.number,
+                    validator: (value) => value!.isEmpty ? requiredAllTerm:null,
+                    controller: _t1Controller,
+                    onTapOutside: (_){
+                      setState(() {
+                        s1 = double.tryParse(_t1Controller.text)!;
```

Comment:
same comment as before. check whether controller text is not empty

Reply:
Done